src/trace/layers/ordered.rs

```diff
@@ -740,6 +740,19 @@ where
         }
     }
 
+    fn seek_last(&mut self) {
+        if self.bounds.1 > self.bounds.0 {
+            self.pos = self.bounds.1 - 1;
+
+            if self.valid() {
+                self.child.reposition(
```

Comment:
I can't figure out whether this is right, perhaps you can write a comment to justify it?

Reply:
Some tests for the function would also be nice